core/src/main/java/org/apache/calcite/sql/validate/implicit/TypeCoercion.java

```diff
@@ -120,6 +120,12 @@ boolean rowTypeCoercion(
    */
   boolean inOperationCoercion(SqlCallBinding binding);
 
+  /** Handles type coercion for Quantify operation {@link org.apache.calcite.sql.fun.SqlQuantifyOperator}.
```

Comment:
Since there are multiple types of quantify operations, I would say "operations" here, and perhaps give an example `SOME(ARRAY [...])` to make it easier to understand.

Does this handle both cases for subqueries and collection constructors? If so, it should be mentioned here as well.

Reply:
Added.